ArrayList/LinkedList.cs

```diff
@@ -4,7 +4,656 @@
 
 namespace List
 {
-    class LinkedList
+    public class LinkedList<T> where T : IComparable<T>
     {
+        public int Length
+        {
+            get
+            {
+                return _length;
+            }
+            set
+            {
+                _length = value >= 0 ? value : 0;
+            }
+        }
+
+        public T this[int index]
+        {
+            get
+            {
+                if (index >= 0 && index < Length)
+                {
+                    Node<T> currentNode = GetNodeByIndex(index);
+
+                    return currentNode.Value;
+                }
+
+                throw new IndexOutOfRangeException();
+            }
+            set
+            {
+                if (index >= 0 && index < Length)
+                {
+                    Node<T> currentNode = GetNodeByIndex(index);
+
+                    currentNode.Value = value;
+                }
+
+                throw new IndexOutOfRangeException();
+            }
+        }
+
+        private int _length;
+        private Node<T> _root;
```

Comment:
Fields first, properties then

Reply:
We agreed to stick to this structure last time.

Reply:
corrected